cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -223,12 +231,20 @@ public void deleteApp(HttpRequest request, HttpResponder responder,
   public void deleteAllApps(HttpRequest request, HttpResponder responder,
                             @PathParam("namespace-id") String namespaceId) throws Exception {
     Id.Namespace id = validateNamespace(namespaceId);
-    try {
-      applicationLifecycleService.removeAll(id);
-      responder.sendStatus(HttpResponseStatus.OK);
-    } catch (CannotBeDeletedException e) {
+    Set<Id.Program> runningPrograms = applicationLifecycleService.removeAll(id);
+    if (!runningPrograms.isEmpty()) {
```

Comment:
Same logic here and in the above method too . Will be nice to refactor and extract the common code as a method.